atlp-rwanda/strikers-bn-be
# Ch/#182193918 configure postgresql and sequelize

#### What does this PR do?

- Set up the migrations and models structure, have a dummy user table and model setup to test the whole configuration

#### Description of Task to be completed?

- User can be able to generate and migrate models, create and run user seed data

#### How should this be manually tested?

- Clone the repo
- Checkout to branch `ch/#182193918-configure-postgresql-and-sequelize`

- Run `npm install --save-dev sequelize-cli` to install sequelize cli
- Change credentials in config/config.json

- Run `npx sequelize-cli model:generate --name User --attributes firstName:string,lastName:string,email:string` to generate a model (you may change the attributes)
- Run `npx sequelize-cli db:migrate` to send the model created above to the database you named in config/config.json
- Run `npx sequelize-cli db:migrate:undo` to revert models sent to the database

Dummy data is also known as seed data. To get some, we generate seed

- Run `npx sequelize-cli seed:generate --name demo-user` and see the new seed under the _seeders_ folder

#### Any background context you want to provide?

- Note that the table name in _seed_ file should be **exactly** the same as that defined in the _model_ file

#### What are the relevant Pivotal Tracker stories?

- [#182215526](https://www.pivotaltracker.com/n/projects/2569362/stories/182215526)

#### Screenshots

- N/A

#### Questions

- N/A
